azure/deploymentframeworks/arm/output.ftl

```diff
@@ -37,19 +37,73 @@
     }]
 [/#function]
 
-[#macro armOutput name type value condition=""]
-    [@mergeWithJsonOutput
-        name="outputs"
-        content=
-            {
-                name : {
-                    "type" : type,
-                    "value" : value
-                } +
-                attributeIfContent("condition", condition)
-            }
-    /]
-[/#macro]
+[#function getArmOutput name type value condition=""]  
+    [#return {
+        name : {
+            "type" : type,
+            "value" : value
+        } +
+        attributeIfContent("condition", condition)
+    }]
+[/#function]
+
+[#function constructArmOutputsFromMappings id name scope mappings=[]]
+    [#local result = {}]
+    [#list mappings as attributeType,attributes]
+        [#local dataType = getOutputMappingDataType(attributeType)]
+        [#list attributes as attributeName,attributeValue]
+
+            [#switch scope]
+
+                [#case "subscription"]
+                [#case "resourceGroup"]
+                    [#local typeFull = getAzureResourceProfile(AZURE_DEPLOYMENT_RESOURCE_TYPE).type]
+                    [#if typeFull?has_content]
+                        [#local propertySections = attributeValue?split(".")]
+                        [#if attributeValue == "id"]
+                            [#local outputName = id]
+                            [#local value = getReference(id, name, typeFull, "", "", "", true, "outputs." + outputName + ".value")]
+                        [#else]
+                            [#local outputName = formatAttributeId(id, propertySections)]
+                            [#local value = getReference(id, name, typeFull, attributeType, "", "", true, "outputs." + outputName + ".value")]
+                        [/#if]
+                    [/#if]
+                    [#break]
+
+                [#case "template"]
+                    [#if attributeValue == "id"]
+                        [#local outputName = id]
+                        [#local value = getReference(id, name)]
+                        [#break]
+                    [#else]
+                        [#local propertySections = attributeValue?split(".")]
+                        [#local outputName = formatAttributeId(id, propertySections)]
+                        [#local typeFull = getAzureResourceProfile(getResourceType(id)).type]
+                        [#if typeFull?has_content]
+                            [#local value = getReference(id, name, typeFull, attributeType, "", "", true, attributeValue)]
+                        [/#if]
+                        [#break]
+                    [/#if]
+                    [#break]
+
+                [#case "pseudo"]
+                    [#return getArmOutput(name, "string", "pseudo")]
+                    [#break]
+
+                [#default]
+                    [@fatal
+                        message=""
```

Comment:
Shouldn't the message have some content? I'd expect an explanation of why its a fatal

Reply:
Ahh yep, I intended to come back to this and must have seen something shiny over yonder

Reply:
he he